src/main/java/seedu/address/model/person/Role.java

```diff
@@ -0,0 +1,63 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Represents a Player's role in League of Legends.
+ * Guarantees: immutable; is valid as declared in {@link #isValidRole(String)}
+ */
+public class Role {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Role must be one of: Top, Jungle, Mid, ADC, Support";
+
+    private static final Set<String> VALID_ROLES = new HashSet<>();
+
+    static {
+        VALID_ROLES.add("Top");
+        VALID_ROLES.add("Jungle");
+        VALID_ROLES.add("Mid");
+        VALID_ROLES.add("ADC");
+        VALID_ROLES.add("Support");
+    }
+
+    public final String roleName;
```

Comment:
```suggestion
    public final RoleType roleType;
```
`RoleType` is just a suggestion for the enum's name.